packages/@tailwindcss-upgrade/src/migrate.ts

```diff
@@ -337,16 +337,54 @@ export async function linkConfigs(
   // All stylesheets have a `@config` directives
   if (withoutAtConfig.length === 0) return
 
-  try {
+  // Find the config file path for each stylesheet
+  let configPathBySheet = new Map<Stylesheet, string>()
+  let sheetByConfigPath = new DefaultMap<string, Set<Stylesheet>>(() => new Set())
+  for (let sheet of withoutAtConfig) {
+    if (!sheet.file) continue
+
+    let localConfigPath = configPath as string
     if (configPath === null) {
-      configPath = await detectConfigPath(base)
-    } else if (!path.isAbsolute(configPath)) {
-      configPath = path.resolve(base, configPath)
+      localConfigPath = await detectConfigPath(path.dirname(sheet.file), base)
+    } else if (!path.isAbsolute(localConfigPath)) {
+      localConfigPath = path.resolve(base, localConfigPath)
     }
 
-    // Link the `@config` directive to the root stylesheets
-    for (let sheet of withoutAtConfig) {
-      if (!sheet.file) continue
+    info(
+      `Linked ${highlight(relative(localConfigPath, base))} to ${highlight(relative(sheet.file, base))}`,
+    )
+    configPathBySheet.set(sheet, localConfigPath)
+    sheetByConfigPath.get(localConfigPath).add(sheet)
+  }
+
+  let problematicStylesheets = new Set<Stylesheet>()
+  for (let sheets of sheetByConfigPath.values()) {
+    if (sheets.size > 1) {
+      for (let sheet of sheets) {
+        problematicStylesheets.add(sheet)
+      }
+    }
+  }
+
+  // There are multiple "root" files without `@config` directives. Manual
+  // intervention is needed to link to the correct Tailwind config files.
+  if (problematicStylesheets.size > 1) {
+    let msg = `You have multiple stylesheets that do not have an \`@config\`.\n`
```

Comment:
Should this include something like "you have multiple stylesheets that we resolved to the same JavaScript config file"? multiple stylesheets alone is not an issue I think, especially when we can find multiple config files 🤔